Artifacts/Get-BCArtifactUrl.ps1

```diff
@@ -141,172 +141,206 @@ try {
         if ($storageAccount -eq 'bcinsider.blob.core.windows.net' -and !$accept_insiderEULA) {
             throw "You need to accept the insider EULA (https://go.microsoft.com/fwlink/?linkid=2245051) by specifying -accept_insiderEula or by providing a SAS token to get access to insider builds"
         }
-        $GetListUrl = "https://$storageAccount/$($Type.ToLowerInvariant())/?comp=list&restype=container"
-    
-        $upMajorFilter = ''
-        $upVersionFilter = ''
-        if ($select -eq 'SecondToLastMajor') {
-            if ($version) {
-                throw "You cannot specify a version when asking for the Second To Last Major version"
-            }
-        }
-        elseif ($select -eq 'Closest') {
-            if (!($version)) {
-                throw "You must specify a version number when you want to get the closest artifact Url"
+
+        if ($type -eq 'sandbox' -and $storageAccount -eq 'bcartifacts.blob.core.windows.net' -and $select -eq 'latest' -and $version -eq '' -and $bcContainerHelperConfig.useApproximateVersion) {
+            # Temp fix / hack for Get-BcArtifact performance
+            # If Microsoft changes versioning schema, this needs to change (or useApproximateVersion should be set to false)
+            $now = ([DateTime]::Now).AddDays(15)
+            $approximateMajor = 23+2*($now.Year-2024)+($now.Month -ge 4)+($now.Month -ge 10)
+            $approximateMinor = ($now.Month - 4)%6
```

Comment:
Won't this become negative in January? What happens when we look for 25.-3?

Reply:
you are right